src/portfolio/APIClient.ts

```diff
@@ -465,7 +471,7 @@ function convertPortfolioDetails(response: DetailsResponse): PortfolioDetails {
       // TODO: remove once API response includes currency
       positions: r.positions.map((p) => ({
         ...p,
-        stock: { ...p.stock, currency: p.stock.currency || '???' },
+        stock: { ...(p.stock || '???') },
       })),
     };
```

Comment:
Why was this changed? Is this function still needed?

Reply:
because p.stock.currency does not exist anymore. the function itself still serves a purpose as it parses the date for nextDividend for example